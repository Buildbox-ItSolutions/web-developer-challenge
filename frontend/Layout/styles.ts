import styled from 'styled-components';

export const Grid = styled.div`
    width: 100%;
    min-height: 100vh;
    display: grid;
    grid-template-columns: 1fr;
    grid-template-rows: 93px auto 100px;
    grid-template-areas:
        'header '
        'main'
        'footer';
`;
