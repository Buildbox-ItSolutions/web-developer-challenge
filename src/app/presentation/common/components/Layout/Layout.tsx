import React from "react";
import { LayoutStyle } from "@/app/presentation/common/components/Layout/styled-components/style";

export interface LayoutInterface {
  children: JSX.Element;
}

const Layout: React.FC<LayoutInterface> = ({ children }) => {
  return <LayoutStyle>{children}</LayoutStyle>;
};

export default Layout;
