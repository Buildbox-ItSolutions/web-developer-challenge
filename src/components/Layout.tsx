import { Outlet } from 'react-router-dom';
import Header from './Header';

export default function Layout() {
  return (
    <>
      <header>
        <Header />
      </header>
      <main>
        <Outlet />
      </main>
    </>
  );
}
