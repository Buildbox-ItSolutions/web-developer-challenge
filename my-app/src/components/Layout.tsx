import { Container } from "react-bootstrap";
import HeaderBar from "./HeaderBar";
import styles from './../../styles/Home.module.scss'
import PostsFeed from "./Post/indext";

export default function Layout({children}:any){
    return(
        <>
        <HeaderBar />
        <Container className={styles.container}>
            <main>{children}</main>
        </Container>
        </>
    )
}