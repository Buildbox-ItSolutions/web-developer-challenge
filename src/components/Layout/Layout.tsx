import * as S from './styles';

type LayoutProps = {
  children: React.ReactNode;
};

export const Layout = ({ children }: LayoutProps) => {
  return <S.Container>{children}</S.Container>;
};
