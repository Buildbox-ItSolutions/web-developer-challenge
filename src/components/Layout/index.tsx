import { ReactNode } from 'react'
import { ThemeProvider } from 'styled-components'
import { StyledContent } from './styles';
import { theme, GlobalStyle } from '../../styles/index'
import Nav from '../Header'

interface ChildrenProps{
    children: ReactNode
}

const Layout = ({children}:ChildrenProps) => {
    return(
        <ThemeProvider theme={theme}>
            <GlobalStyle />
            

            <StyledContent>
                <Nav />

                <div id="content">
                    {children}
                </div>
            </StyledContent>
        </ThemeProvider>
    )
}

export default Layout;