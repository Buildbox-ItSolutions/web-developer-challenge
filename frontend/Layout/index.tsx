import { FC } from 'react';

import { Footer, Header, Main } from '../components';
import { Grid } from './styles';

const Layout: FC = () => {
    return (
        <Grid data-test="layout-grid-component">
            <Header data-test="layout-header-component" />
            <Main data-test="layout-main-component" />
            <Footer data-test="layout-footer-component" />
        </Grid>
    );
};

export default Layout;
