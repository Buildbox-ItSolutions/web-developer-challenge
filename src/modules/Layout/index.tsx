import { Outlet, useNavigate } from "react-router-dom";
import { GlobalStyle } from "../../styles/GlobalStyle";
import { Header } from "../../components";

export const Layout = () => {
  const navigate = useNavigate();
  navigate("./feed");

  return (
    <>
      <GlobalStyle />
      <Header />
      <Outlet />
    </>
  );
};
