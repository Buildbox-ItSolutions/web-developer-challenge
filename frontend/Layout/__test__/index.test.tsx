import 'jest-styled-components';

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Layout from '../index';

describe('Layout', () => {
    it('must render without errors', () => {
        const wrapper = renderer.create(<Layout />).toJSON();

        expect(wrapper).toMatchSnapshot();
    });

    it('should display the layout with grid, header, main, footer.', () => {
        const wrapper = shallow(<Layout />);
        expect(wrapper.find(`[data-test='layout-grid-component']`).exists()).toBeTruthy();
        expect(wrapper.find(`[data-test='layout-header-component']`).exists()).toBeTruthy();
        expect(wrapper.find(`[data-test='layout-main-component']`).exists()).toBeTruthy();
        expect(wrapper.find(`[data-test='layout-footer-component']`).exists()).toBeTruthy();
    });
});
