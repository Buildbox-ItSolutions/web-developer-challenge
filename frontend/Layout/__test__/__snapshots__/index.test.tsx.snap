// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout must render without errors 1`] = `
.c1 {
  width: 100%;
  height: 93px;
  grid-area: header;
  background-color: #2b2b2b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: fixed;
  z-index: 100;
}

.c2 {
  width: 103px;
  height: 45px;
}

.c4 {
  width: 80%;
  height: 353px;
  background-color: #313131;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 3px;
  padding: 2em;
  border: 1px solid #3b3b3b;
  grid-column-start: 2;
  grid-column-end: 2;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  margin-right: 1em;
  width: 88px;
  height: 88px;
  border-radius: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border: 1px solid #4b4b4b;
}

.c6 input {
  display: none;
}

.c6 label {
  z-index: 10;
  width: 100%;
  height: 100%;
  border-radius: 36px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  text-align: center;
}

.c8 {
  width: 100%;
  border-radius: 8px;
  background-color: #494949;
  border: none;
  color: #ffff;
  padding: 0.9em;
  height: 40px;
}

.c8:focus {
  outline-width: 0;
}

.c9 {
  width: 100%;
  border-radius: 8px;
  background-color: #494949;
  border: none;
  color: #ffff;
  padding: 0.9em;
  height: 80px;
  resize: none;
}

.c9:focus {
  outline-width: 0;
}

.c10 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c12 {
  width: 98px;
  height: 41px;
  object-fit: contain;
  border-radius: 8px;
  background-color: #5f5f5f;
  color: #313131;
  border: none;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  font-size: 14px;
  cursor: pointer;
}

.c12:focus {
  outline-width: 0;
}

.c12:active {
  background-color: #71bb00;
  color: #fff;
}

.c11 {
  width: 60px;
  font-size: 14px;
  color: #5f5f5f;
  background: none;
  border: none;
  cursor: pointer;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  margin-right: 1.5em;
}

.c11:focus {
  outline-width: 0;
}

.c7 {
  width: 24px;
  height: 24px;
  cursor: pointer;
  border-radius: 36px;
}

.c3 {
  padding-top: 20px;
  width: 100%;
  height: auto;
  grid-area: main;
  background-color: #343434;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c13 {
  width: 100%;
  height: 100px;
  color: #7a7a7a;
  font-size: 14px;
  grid-column-start: 2;
  grid-column-end: 2;
  display: grid;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-bottom: 10px;
}

.c14 {
  width: 100%;
  height: 100px;
  grid-area: footer;
  background-color: #2b2b2b;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c15 {
  color: #9f9f9f;
  font-size: 1em;
}

.c0 {
  width: 100%;
  min-height: 100vh;
  display: grid;
  grid-template-columns: 1fr;
  grid-template-rows: 93px auto 100px;
  grid-template-areas: 'header ' 'main' 'footer';
}

@media (min-width:900px) {
  .c4 {
    width: 100%;
  }
}

@media (min-width:900px) {
  .c3 {
    display: grid;
    grid-template-rows: auto auto auto;
    grid-template-columns: 1.5fr 2fr 1.5fr;
  }
}

@media (max-width:900px) {
  .c13 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

@media (max-width:342px) {
  .c14 {
    text-align: center;
  }
}

<div
  className="c0"
  data-test="layout-grid-component"
>
  <header
    className="c1"
  >
    <img
      alt="Logo"
      className="c2"
      data-test="header-logo"
      src="/bx-logo.png"
    />
  </header>
  <main
    className="c3"
  >
    <form
      className="c4"
      data-test="publisher-form"
      onSubmit={[Function]}
    >
      <div
        className="c5"
      >
        <div
          className="c6"
        >
          <label
            htmlFor="file"
          >
            <img
              alt="profile"
              className="c7"
              height="24px"
              src="image.svg"
              width="24px"
            />
          </label>
          <input
            id="file"
            name="file"
            onChange={[Function]}
            type="file"
          />
        </div>
      </div>
      <input
        className="c8"
        data-test="publisher-input-name"
        name="name"
        onChange={[Function]}
        placeholder="Digite seu nome"
        required={true}
        value=""
      />
      <textarea
        className="c9"
        data-test="publisher-textarea-msg"
        maxLength={160}
        name="message"
        onChange={[Function]}
        placeholder="Mensagem"
        required={true}
        value=""
      />
      <div
        className="c10"
      >
        <button
          className="c11"
          onClick={[Function]}
          type="reset"
        >
          Descartar
        </button>
        <button
          className="c12"
          data-test="publisher-btn-submit"
          type="submit"
        >
          Publicar
        </button>
      </div>
    </form>
    <p
      className="c13"
    >
      Feed
    </p>
  </main>
  <footer
    className="c14"
  >
    <p
      className="c15"
    >
      Â© Copyright 2019 - 2020.
    </p>
    <p
      className="c15"
    >
      All rights reserved. Powered by the 
      <strong>
        Buildbox
      </strong>
    </p>
  </footer>
</div>
`;
